iac-extensions/jvm-framework-config/src/main/java/org/sonar/iac/jvmframeworkconfig/parser/yaml/JvmFrameworkConfigYamlConverter.java

```diff
@@ -68,7 +69,11 @@ private Profile convertToProfile(Node node) {
     commentsPerProfile = new ArrayList<>();
     List<Tuple> tuples = convert(node).map(TupleBuilder::build).toList();
 
-    return new ProfileImpl(tuples, commentsPerProfile, profileName(tuples), true);
+    var profile = new ProfileImpl(tuples, commentsPerProfile, profileName(tuples), true);
+    // This step is needed for syntax highlighting later. Keeping track of YAML nodes in the leaves would be more complicated,
+    // because in the flattened structure leaves would have a lot of common ancestors, and they would need to be filtered later.
+    profile.setOriginalYamlTree(yamlConverter.convert(node));
```

Comment:
An alternative I was considering: store all key nodes in all leaf nodes (i.e., in `Tuple`s), and inside the highlighting visitor get all these nodes for a profile, filter out duplicates, and highlight. This will require more code, and it's not obvious whether it will be cheaper than an additional conversion here

Reply:
Do you have a rough estimate how long this conversion `yamlConverter.convert(node)` is taking for files? 
Having a measurement would help. We don't need it ultra detailed, just a rough estimate would be nice.

Reply:
From running ITs locally:
```
INFO: 1265/1265 source files have been analyzed
INFO: Duration Statistics, Parse 372 ms, JvmFrameworkConfigMetricsVisitor 239 ms, JvmFrameworkConfigHighlightingVisitor 219 ms, ChecksVisitor 103 ms, **yaml-conversion 30 ms**, S6437 21 ms, S4423 12 ms, S1135 6 ms, S3330 3 ms, S2092 2 ms, S4830 2 ms, S5693 2 ms, S4507 1 ms
```
**`yaml-conversion 30 ms`**
So I'd say quite fast

Reply:
That sounds fast enough.

Reply:
I believe you choose the right approach here.
The only thing I was wondering if it makes sense to put it under File and not Profile.
Was there a specific reason you choose this approach?

Reply:
Not really; I don't have a strong opinion, and probably putting it under File would reduce the number of entities that are aware of the original tree